### PR TITLE
Implement 3rd-party bundle-list

### LINKED ISSUE #1179: Create a command to list all bundles from 3rd-party repositories
Create a command `swupd 3rd-party bundle-list` that can list bundles from a 3rd party repository

The goal is to support all features from bundle-list that are supported in `swupd bundle-list`, but focus should be on:

`swupd 3rd-party bundle-list`: List all installed bundles grouped by repository
`swupd 3rd-party bundle-list --repo my_repo`: List all installed bundles from my_repo
`swupd 3rd-party bundle-list -all`: List all installed bundles grouped by repository
`swupd 3rd-party bundle-list --all --repo my_repo`: List all bundles from my_repo

Note that we should support installation of multiple bundles with the same name. If there's any name clash the repository name should be required.

Related #1082

Tasks:
- [x] Add support for `3rd-party bundle-list` to the `testlib`
- [x] Add functional tests for listing bundles from a 3rd-party repos
- [x] Add the `3rd-party bundle-list` command to swupd (without functionality)
- [x] Add the `--repo` flag to specify a 3rd-party repo
- [ ] Add documentation to man pages.
- [x] Add command to autocompletion script swupd.bash
- [x] Add command to autocompletion script swupd.zsh
- [x] Add command to the config file
- [x] Implementation of the `3rd-party bundle-list` functionality
- [x] Implementation of the `--repo` flag functionality
- [x] Check output correctness and consistency with other commands (including output using `--quiet`, `--verbose`, `--debug`)
- [ ] Update release notes draft

### PR DESCRIPTION
**Pull request by castulo**
Implementation of the `swupd 3rd-party bundle-list` command.

Signed-off-by: Castulo Martinez <castulo.martinez@intel.com>

Closes #1179

### CODE REVIEW

File: src/bundle.c [hunk 1]
```diff
@@ -158,7 +158,8 @@ int required_by(struct list **reqd_by, const char *bundle_name, struct manifest
 					print_msg = false;
 					info("%s", msg);
 				}
-				info(" - %s\n", name);
+				info(" - ");
```
Comment (otaviobp) [inline src/bundle.c]: Separate bug fixes or other improvements from main commit. It's hard to understand why this change is made here

File: src/bundle_list.c [hunk 2]
```diff
@@ -151,21 +147,25 @@ static enum swupd_code list_local_bundles()
 		return SWUPD_COULDNT_LIST_DIR;
 	}
 
+	info("Installed bundles:\n");
```
Comment (otaviobp) [inline src/bundle_list.c]: You are changing one feature on list-bundles that is unrelated to 3rd_party. Do that in a separate commit.

File: test/functional/3rd-party/3rd-party-bundle-list-basic.bats [hunk 3]
```diff
@@ -45,16 +78,35 @@ test_setup() {
...
+		Total: 2
+	EOM
+	)
+	assert_is_output --identical "$expected_output"
 
 }
 
-@test "TPR020: List installed 3rd-party bundles from a specific repo" {
 
+@test "TPR020: List installed 3rd-party bundles from a specific repo" {
```
Comment (otaviobp) [inline test/functional/3rd-party/3rd-party-bundle-list-basic.bats]: Don't add a test in a PR if it's not being covered.

File: test/functional/3rd-party/3rd-party-bundle-list-deps.bats [hunk 4]
```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bats
+
+# Author: Castulo Martinez
+# Email: castulo.martinez@intel.com
+
+load "../testlib"
+
+global_setup() {
```
Comment (otaviobp) [inline test/functional/3rd-party/3rd-party-bundle-list-deps.bats]: Why are some tests added in the commit: "Add basic tests for 3rd-party bundle-list" and some tests are added in "Implementing 3rd-party bundle-list"?

File: src/3rd_party_bundle_list.c [hunk 5]
```diff
@@ -107,9 +107,41 @@ static bool parse_options(int argc, char **argv)
 	return true;
 }
 
+static void print_repo_header(const char *repo_name)
+{
+	const int HEADER = 18;
```
Comment (otaviobp) [inline src/3rd_party_bundle_list.c]: where's this 18 came from? Can't it be replaced by a sizeof()?
magic numbers are usually hard to maintain.

Comment (castulo) [reply src/3rd_party_bundle_list.c]: It is the length of " 3rd-Party Repo:  ", I'll modify this so we don't have magic numbers

File: src/bundle.c (hunk 1 above)
Comment (castulo) [reply src/bundle.c]: ok, will do

File: src/bundle_list.c (hunk 2 above)
Comment (castulo) [reply src/bundle_list.c]: 👍

**Comment by castulo** [comment]
V2:
- Removed magic number from print_repo_header()
- split changes unrelated to 3rd_party bundle-list implementation to a separate commit
- removed tests related to code that is not yet being implemented from this PR
